src/Runner.Worker/JobExtension.cs

```diff
@@ -717,33 +739,58 @@ public async Task FinalizeJob(IExecutionContext jobContext, Pipelines.AgentJobRe
             }
         }
 
-        private async Task<string> CheckConnectivity(string endpointUrl)
+        private async Task<CheckResult> CheckConnectivity(string endpointUrl, string accessToken, int timeoutInSeconds, CancellationToken token)
```

Comment:
I make this more generic so i can reuse between the current 1 time connection check and the backend long-running connection check.